File: 0797-rabbits-in-forest/0797-rabbits-in-forest.go
func numRabbits(answers []int) int {
    freq := make(map[int]int)    
    for _, a := range answers {
        freq[a]++
    }

    res := 0
    for k, v := range freq {
        groupSize := k+1
        groups := (v+k)/groupSize
        res += groupSize * groups
    }
    return res
}

File: 0797-rabbits-in-forest/0797-rabbits-in-forest.cpp
using namespace std;

class Solution {
public:
    int numRabbits(vector<int>& answers) {
        unordered_map<int, int> freq;
        int res = 0;

        for (int a : answers) {
            freq[a]++;
        }

        for (auto& [x, count] : freq) {
            int group_size = x + 1;
            int groups = (count + x) / group_size;
            res += groups * group_size;
        }

        return res;
    }
};